graphkit/network.py

```diff
@@ -349,3 +427,50 @@ def get_node_name(a):
             plt.show()
 
         return g
+
+
+def ready_to_schedule_operation(op, has_executed, graph):
+    """
+    Determines if a Operation is ready to be scheduled for execution based on
+    what has already been executed.
+
+    Args:
+        op:
+            The Operation object to check
+        has_executed: set
+            A set containing all operations that have been executed so far
+        graph:
+            The networkx graph containing the operations and data nodes
+    Returns:
+        A boolean indicating whether the operation may be scheduled for
+        execution based on what has already been executed.
+    """
+    dependencies = set(v for v in nx.ancestors(graph, op) if isinstance(v, Operation))
```

Comment:
same thing, but more explicit, that it's only filtering.
```
dependencies = set(filter(lambda v: isinstance(v, Operation), nx.ancestors(graph, op))
```

Reply:
okay i'll update.